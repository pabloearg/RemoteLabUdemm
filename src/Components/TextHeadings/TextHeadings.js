// @flow
import React from 'react';
import { Text } from 'react-native';
import type { TextStyle } from 'react-native';
import { TextStyles } from '../../styles';
import type { HEADING_TYPES } from '../../styles/text';
/* eslint-disable react/jsx-props-no-spreading */
type Props = {
  type?: HEADING_TYPES,
  children?: any,
  color?: "black" | "white" | undefined,
  styleText?: TextStyle,
  text?: string,
}

const getStyle = (type: HEADING_TYPES) => TextStyles[type] ?? TextStyles.h4;

const TextHeadings = ({
  type,
  children,
  styleText,
  color,
  text,
  ...rest
}: Props) => (
  <Text
    {...rest}
    style={[
      TextStyles.fontFamily,
      getStyle(type),
      styleText,
      color && { color }]}
  >
    {text ?? children}

  </Text>
);

TextHeadings.defaultProps = {
  color: undefined,
  type: 'h4',
  text: undefined,
  children: undefined,
  styleText: undefined,
};

export default TextHeadings;
