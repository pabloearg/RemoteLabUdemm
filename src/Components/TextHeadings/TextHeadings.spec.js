import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TextHeadings from '.';

describe('TextHeadings', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(
        <TextHeadings type="h1">NEXT</TextHeadings>
      );
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
