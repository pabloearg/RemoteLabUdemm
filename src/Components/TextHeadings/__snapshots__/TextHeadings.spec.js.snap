// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextHeadings Rendering should match to snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "avenir",
      },
      Object {
        "fontSize": 32,
        "fontWeight": "900",
        "lineHeight": 44,
      },
      undefined,
      undefined,
    ]
  }
>
  NEXT
</Text>
`;
